lib/colorado_lottery.rb

```diff
@@ -0,0 +1,60 @@
+class ColoradoLottery
+
+  attr_reader :registered_contestants, :winners, :current_contestants
+  def initialize
+    @registered_contestants = Hash.new { |hash, key| hash[key] = [] }
+    @winners = []
+    @current_contestants = Hash.new { |hash, key| hash[key] = [] }
+  end
+
+  def interested_and_18?(contestant, game)
+    contestant.game_interests.include?(game.name) && contestant.age >= 18
+  end
+
+  def can_register?(contestant, game)
+    interested_and_18?(contestant, game) &&
+    (!contestant.out_of_state? || game.national_drawing?)
+  end
+
+  def register_contestant(contestant, game)
+
+    registered_contestants[game.name] << contestant if can_register?(contestant,game)
+  end
+
+  def eligible_contestants(game)
+    registered_contestants[game.name]
+  end
+
+  def charge_contestants(game)
+    eligible_contestants(game).each do |contestant|
```

Comment:
Solid, exactly what we wanted to see, but because `eligible_contestants` did not have the `find_all` some contestants were populated in the `current_contestants` hash that should not have been.